generator/test_case_generator.go

```diff
@@ -31,6 +31,7 @@ type matrixRow struct {
 	TerraformDir        string `json:"terraform_dir"`
 	UseSSM              bool   `json:"useSSM"`
 	ExcludedTests       string `json:"excludedTests"`
+	MetadataEnabled     string `json:"metadataEnabled"`
```

Comment:
Will this impact other tests where the parameter is not specified? I presume we want to have this enabled by default nearly everywhere to preserve the current tests.

Reply:
https://github.com/aws/amazon-cloudwatch-agent/actions/runs/5863584354/job/15897572781 
None of the other tests use this variable so it should not affect them. I ran a full test just in case and it does not show a change of behavior for other tests. 

Note: Failed tests are due to multi-config issues.